pygac_fdr/tests/test_end2end.py

```diff
@@ -0,0 +1,470 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2020 pygac-fdr developers
+#
+# This file is part of pygac-fdr.
+#
+# pygac-fdr is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# pygac-fdr is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# pygac-fdr. If not, see <http://www.gnu.org/licenses/>.
+
+"""End-to-end tests for pygac-fdr.
+
+Download test data set, run the entire chain of processing steps (read GAC files, write data
+to netCDF, enhance metadata) and compare results against reference data. They should be identical.
+
+Usage:
+
+$ python test_end2end.py
+
+The test behaviour can be controlled using the configuration file test_end2end.yaml .
+
+Do not use pytest as this will eat up all your memory, because pytest caches all generated figures
+until the end of the test suite for some reason.
+"""
+
+from cfchecker.cfchecks import CFChecker
+from dateutil.parser import isoparse
+import glob
+import gzip
+import logging
+import numpy as np
+import os
+import shutil
+import subprocess
+import unittest
+import xarray as xr
+from xarray.core.utils import dict_equiv
+from xarray.core.formatting import diff_attrs_repr
+import yaml
+
+from pygac_fdr.metadata import QualityFlags
+from pygac_fdr.utils import logging_on, LOGGER_NAME
+
+
+LOG = logging.getLogger(LOGGER_NAME)
+
+
+def call_subproc(cmd, cwd='.'):
+    LOG.info('Calling subprocess {}'.format(cmd))
+    ret = subprocess.call(cmd, cwd=cwd)
```

Comment:
Consider using `subprocess.run` with the check option instead of raising the exception manually. See example in https://docs.python.org/3/library/subprocess.html#subprocess.run.